### DISCUSSION TITLE
使用unplugin-vue-components如何规避打包icons？

### DISCUSSION THREAD

**Discussion by Rafong973** [element-plus/element-plus#6238] [Q&A]
开发者你们好，我使用了unplugin-vue-components做了按需加载方案，我在项目中并无使用icons，但打包的时候icons仍会被打包进去，我还使用了unplugin-icons，但发现还是icons还是全量打包。最后我删掉了node_modules中的@element-plus/icons-vue，发现报错信息是来自element-plus/es/components/backtop等组件。请问这个问题该如何解决？

<img width="1806" alt="iShot2022-02-23 14 32 31" src="https://user-images.githubusercontent.com/22493242/155271415-f74d0766-ecc1-461b-a3e3-36a855ab4d02.png">
<img width="1173" alt="iShot2022-02-23 14 32 38" src="https://user-images.githubusercontent.com/22493242/155271432-264291c6-9744-4f1e-8aae-42eb6fd024ca.png">

**Comment by sxzz** [comment]
Webpack吗？是否开启 Tree shaking 相关配置？

**Comment by sxzz** [comment]
仔细看了下，Element Plus 本身内部用到了部分 icons，正常来说只会引入用了的icon。你的截图左边那块都是Element Plus本身组件。建议给出复现仓库。另外你的按需加载方案似乎有问题，看截图似乎是全量加载。

**Reply by sxzz** [reply]
BTW, webpack 需要在 `production` 的模式下看打包出来的体积才有参考价值。

**Comment by vikingmute** [comment]
开发者们你们好 我也遇到相同问题，请看截图
<img width="981" alt="屏幕快照 2022-07-13 上午11 54 04" src="https://user-images.githubusercontent.com/343954/178647352-66893b6f-a1f7-48ee-aeb6-2c2fceb4ead4.png">
脚手架工具为 vue-cli，tree shaking 是开启的，现在是 Production build，从左侧可以看到按需加载已经开启。项目中并没有使用任何的 ICON，图标是来自 element-plus 的某些组件，但是 tree shaking 并没有实现，直接加载了 cjs 文件。Webpack 版本为 4.46.0。供参考。

**Reply by sxzz** [reply]
请尝试 Webpack 5

**Reply by vikingmute** [reply]
非常感谢 今天试了 webpack5 没问题了～
